validation/validate.py

```diff
@@ -0,0 +1,268 @@
+    embeddings1 = embeddings[0::2]
+    embeddings2 = embeddings[1::2]
+    accuracy, recall, precision = _score_k_fold(thresholds,
+                                                embeddings1,
+                                                embeddings2,
+                                                labels,
+                                                num_folds,
+                                                distance_metric,
+                                                subtract_mean,
+                                                divide_stddev)
+    return np.mean(accuracy), np.mean(recall), np.mean(precision)
+
+
+def _read_pairs(pairs_filename: str) -> List[Pair]:
+    pairs = []
+    with open(pairs_filename, 'r') as pair_file:
+        for line_num, line in enumerate(pair_file):
+            if line_num != 0:
+                pair = cast(Pair, tuple([int(i) if i.isdigit() else i
+                                         for i in line.strip().split()]))
+                pairs.append(pair)
+    return pairs
+
+
+def _get_paths_and_labels(image_dir: str,
+                          pairs: List[Pair]) -> Tuple[List[Path], List[Label]]:
+    paths = []
+    labels = []
+    for pair in pairs:
+        _add_extension = (lambda rel_image_path, image_dir:
+                          f'{rel_image_path}.jpg'
+                          if exists(join(image_dir, f'{rel_image_path}.jpg'))
+                          else f'{rel_image_path}.png')
+        if len(pair) == 3:
+            person, image_num_0, image_num_1 = cast(Match, pair)
+            rel_image_path_no_ext = join(person,
+                                         f'{person}_{image_num_0:04d}')
+            rel_image_path_0 = _add_extension(rel_image_path_no_ext, image_dir)
+            rel_image_path_no_ext = join(person,
+                                         f'{person}_{image_num_1:04d}')
+            rel_image_path_1 = _add_extension(rel_image_path_no_ext, image_dir)
+            is_same_person = True
+        elif len(pair) == 4:
+            person_0, image_num_0, person_1, image_num_1 = cast(Mismatch, pair)
+            rel_image_path_no_ext = join(person_0,
+                                         f'{person_0}_{image_num_0:04d}')
+            rel_image_path_0 = _add_extension(rel_image_path_no_ext, image_dir)
+            rel_image_path_no_ext = join(person_1,
+                                         f'{person_1}_{image_num_1:04d}')
+            rel_image_path_1 = _add_extension(rel_image_path_no_ext, image_dir)
+            is_same_person = False
+        if (exists(join(image_dir, rel_image_path_0))
+                and
+                exists(join(image_dir, rel_image_path_1))):
+            paths.append((rel_image_path_0, rel_image_path_1))
+            labels.append(is_same_person)
+        else:
+            err = f'{rel_image_path_no_ext} with .jpg or .png extensions'
+            raise FileNotFoundError(err)
```

Comment:
In this case, why not skip the entry with an error message to make the life easier for the user?